modules/plugin/spellcheck/src/main/java/net/java/sip/communicator/plugin/spellcheck/Parameters.java

```diff
@@ -51,14 +51,13 @@ class Parameters
     {
         try
         {
-            URL url =
-                SpellCheckActivator.bundleContext.getBundle().getResource(
-                    RESOURCE_LOC);
-
-            InputStream stream = url.openStream();
+            final InputStream stream = Parameters.class.getClassLoader()
```

Comment:
This case is unique and unnecessary use: it goes back to the bundle and then stays within the bundle. I agree that it would be nicer if a total OSGi implementation was available, but how often do you load resources across bundle boundaries? I suspect it is rare, and this case was false, so not even an occurrence.

Reply:
@ibauersachs to add, I hadn't dismissed your remark about fixing the custom OSGi. I am worried that the lack of interest in the desktop app may not warrant investing that much time. (That is the impression I am getting.) You mentioned before that we might simply move away from the more elaborate uses of OSGi and stick with plugin management.

I don't know enough about jitsi's user base or its OSGi use, so feel free to correct me. Is this more than a gut feeling for you? Like, do you have a list of (problematic) uses? Or something else actionable?

Reply:
Well, any time put into Jitsi is probably just for fun or to learn something new by now.

No, I don't. The reason I moved away from full OSGi was that too many libraries don't have an OSGi manifest and all the problems that come with it (like rebundling). Also, it's way faster now. The drawback is of course that Jitsi was relying on real OSGi and stuff might silently be broken, like the spellchecker you found.